src/main/java/edu/ucla/library/avpairtree/verticles/MainVerticle.java

```diff
@@ -135,17 +139,30 @@ private void configureServer(final JsonObject aConfig, final Promise<Void> aProm
                     futures.add(deployVerticle(new WaveformVerticle(), aConfig.copy().put(WORKER, true)));
 
                     CompositeFuture.all(futures).onSuccess(result -> {
-                        startCsvDirWatcher(aConfig).onComplete(startup -> {
-                            // Register the codec for passing CsvItem(s) over the event bus
-                            vertx.eventBus().registerDefaultCodec(CsvItem.class, new CsvItemCodec());
-
-                            if (startup.succeeded()) {
-                                LOGGER.info(MessageCodes.AVPT_001, port); // Log a successful startup w/ port
-                                aPromise.complete();
+                        try {
+                            // Configure the waveform consumer
```

Comment:
I'm wondering if it wouldn't be better to have one waveform consumer that hits different sources based on configuration rather than two versions of it. With two versions, in the tests, then your testing the test (localstack) consumer and not the real consumer, right? I'll ask questions in slack since I may be missing something here.

Reply:
I think so too. Initially, there wasn't as much shared code, but now the two classes are almost the same. Will do.

Reply:
Good call on this, I hadn't realized just how much the two classes had converged over time.